File: app/src/main/java/name/monwf/customiuizer/CredentialsShortcut.java
package name.monwf.customiuizer;

import android.content.Intent;
import android.os.Bundle;

import androidx.appcompat.app.AppCompatActivity;

public class CredentialsShortcut extends AppCompatActivity {

	@Override
	protected void onCreate(Bundle savedInstanceState) {
		super.onCreate(savedInstanceState);

		Intent intent = new Intent();
		Intent launchIntent = new Intent(this, Credentials.class);
		intent.putExtra(Intent.EXTRA_SHORTCUT_INTENT, launchIntent);
		intent.putExtra(Intent.EXTRA_SHORTCUT_NAME, getResources().getString(R.string.credentials_unlock));
		intent.putExtra(Intent.EXTRA_SHORTCUT_ICON_RESOURCE, Intent.ShortcutIconResource.fromContext(this, R.drawable.ic_credentials));

		setResult(RESULT_OK, intent);
		finish();
	}

}

File: app/src/main/java/tv/withaibuild/customiuizer/CredentialsShortcut.kt
package tv.withaibuild.customiuizer

import android.content.Intent
import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity

class CredentialsShortcut : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

        val launchIntent = Intent(this, Credentials::class.java)
        val intent = Intent().apply {
            putExtra(Intent.EXTRA_SHORTCUT_INTENT, launchIntent)
            putExtra(Intent.EXTRA_SHORTCUT_NAME, getString(R.string.credentials_unlock))
            putExtra(Intent.EXTRA_SHORTCUT_ICON_RESOURCE, Intent.ShortcutIconResource.fromContext(this@CredentialsShortcut, R.drawable.ic_credentials))
        }

        setResult(RESULT_OK, intent)
        finish()
    }
}
